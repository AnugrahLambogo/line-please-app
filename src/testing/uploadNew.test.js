import React from 'react';
import ReactDOM from 'react-dom';
import UploadNew from '../uploadNew';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <UploadNew />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <UploadNew />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});