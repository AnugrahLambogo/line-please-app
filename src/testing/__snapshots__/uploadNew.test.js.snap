// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section>
  <h3>
    Upload new oration
  </h3>
  <p>
    Write or copy/paste the text you want to memorize below. Separate your text by 'sections' as you see fit, and make sure each 'line' is separated by an enter/return.
  </p>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="textTitle"
      id="titleLabel"
    >
      Title:
    </label>
    <input
      id="textTitle"
      onChange={[Function]}
      type="text"
    />
    <section
      className="uploadForm"
    >
      <label
        className="sectionLabel"
        htmlFor="section1"
      >
         Section 
        1
        :
      </label>
      <textarea
        className="sections"
        name="section1"
        onChange={[Function]}
        type="text"
      />
    </section>
    <section
      className="uploadForm"
    >
      <label
        className="sectionLabel"
        htmlFor="section2"
      >
         Section 
        2
        :
      </label>
      <textarea
        className="sections"
        name="section2"
        onChange={[Function]}
        type="text"
      />
    </section>
    <section
      className="uploadForm"
    >
      <label
        className="sectionLabel"
        htmlFor="section3"
      >
         Section 
        3
        :
      </label>
      <textarea
        className="sections"
        name="section3"
        onChange={[Function]}
        type="text"
      />
    </section>
    <button
      className="uploadButton"
      onClick={[Function]}
    >
      Add another section
    </button>
    <button
      className="uploadButton"
      onClick={[Function]}
    >
      Remove section
    </button>
    <button
      className="uploadButton"
      type="submit"
    >
      Create!
    </button>
  </form>
</section>
`;
